gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallableFactory.java

```diff
@@ -83,7 +86,22 @@ private GrpcCallableFactory() {}
    */
   public static <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> createDirectCallable(
       MethodDescriptor<RequestT, ResponseT> methodDescriptor) {
-    return new GrpcDirectCallable<>(methodDescriptor);
+    return createDirectCallable(methodDescriptor, EmptyRequestParamsExtractor.<RequestT>of());
+  }
+
+  /**
+   * Create a callable object that directly issues the call to the underlying API with nothing
+   * wrapping it. Designed for use by generated code.
+   *
+   * @param methodDescriptor the gRPC method descriptor
+   * @param paramsExtractor request message parameters extractor, which will be used to populate
+   *     routing headers
+   */
+  public static <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> createDirectCallable(
```

Comment:
Add a special `@BetaApi` marker to this API (in case of emergency, removing `@BetaApi` from the class still would retain it here)